### ISSUE TITLE
Feature Request: Dynamically Set Slack Channel

### ISSUE THREAD

**Issue by connoranderson** [influxdata/kapacitor#2137]
This issue is derived from: https://github.com/influxdata/kapacitor/issues/1404

Slack alert node currently supports setting channel in a given TICK script. However, it is currently not able to dynamically set the channel based on a tag or different field in an input measurement. It would definitely be a useful enhancement to have more flexibility to for this channel accept template inputs.

**Comment by nativb** [comment]
Is this fixed?

The example in the docs does exactly that, but it looks like it doesn't work

https://docs.influxdata.com/kapacitor/v1.5/working/template_tasks/

**Comment by erkexzcx** [comment]
Related: https://github.com/influxdata/kapacitor/issues/2283#issuecomment-790618842

Since this issue is from 2018, I think we should wait to see what InfluxDB 2.* would bring in regards to alerting. So far latest influxdb 2.* builds support only few alerting endpoints (e.g. Slack), but not Telegram which is deal breaker to me.

**Comment by leonty** [comment]
Yeah, same problems with telegram's .chatId() method. The problem is that it's not possible to set dynamically the parameter because of incapacity of tick script. Its syntax or the ways it's being interpreted should be changed to make it possible to use normal programming techniques like passing a variable value instead of hardcoded string.
